### ISSUE TITLE
No color settings in VS 2015

### ISSUE THREAD

**Issue by Lex-DRL** [samizzo/nshader#15]
The title is self-explanatory.
Just installed the latest release (2.4) of this ext to the VS 2015 - and there's no syntax highlighting at all. More importantly, no way to specify it manually because there are no "NShader – Type" settings under Tools > Options > Environment > Fonts and Colors.

Tried uninstalling and re-instaloing the ext, also tried version prior to VS 2017 support (2.1) - the same thing.

**Comment by samizzo** [comment]
Did you have a previous version of NShader installed, or any other shader syntax highlighters?

**Comment by Lex-DRL** [comment]
I didn't have NShader installed in the current version of VS _(2015 update 3)_.
But I may have tried it some time ago, when I was using VS 2013. 
Anyway, prior to the installation of my currently used VS, I have manually cleaned up whatever I was able to find _(any tracks of VS in Program Files, AppData and ProgramData, including Common* folders)_.

However, I tried the following extension _(installed and uninstalled it)_ before installing your version of NShader:
https://marketplace.visualstudio.com/items?itemName=MarcinODev.ShaderUnitySupport

**Comment by Lex-DRL** [comment]
I can try installing your extension on the "pure" VS, if you tell me how I can do that. What files/folders on my system disk I need to remove to clean up VS for this ext?

**Comment by samizzo** [comment]
I'm not totally sure what to do aside from just uninstalling the extension from within VS. You could try extrapolating the instructions [here](http://stackoverflow.com/questions/25234634/how-to-uninstall-a-visual-studio-2013-extension-manually) to NShader.

**Comment by Lex-DRL** [comment]
That topic is about another issue: uninstallig ah extension when it can't do it properly from GUI in VS. I don't have this issue. I have successfully uninstalled that extention before installing yours.

It may have left some of it's settings somewhere in the OS, but I don't know where (if it did).

**Comment by Lex-DRL** [comment]
I've spent entire day uninstalling VS, .NET framework and VC++ redistributables and then cleaning my system from any garbage left after it. I've removed everything I could find (both on the system disk and in the registry). I searched for any mentions of VS, NSHADER, just "shader" and Resharper.

After it, I've installed from scratch latest .NET framework + VS 2017 (Community).
And now, on the perfectly clean VS, I'm installing your version of NSHADER.

But still, there are no color settings.
I do have "Shader Language" options block under "Text Editor" group. But there's not even a single option labled "NSHADER - ..." under "Environment" group.

I'm desparate.
Are you sure it works in VS 2015/2017?
![001](https://cloud.githubusercontent.com/assets/3897975/26286444/55e78926-3e6e-11e7-95c7-5ab44eea15e4.png)
![002](https://cloud.githubusercontent.com/assets/3897975/26286447/6ae3d136-3e6e-11e7-8e46-2350cc1bdea7.png)

**Comment by samizzo** [comment]
I'm sorry to hear that. I've been using it in VS2015 and 2017 on my home machine, and other people have reported that it works as well. However, I have just installed VS2017 on my work machine and now I see that it's not working, so at least I have a repro case now! I haven't yet figured out what's going on; the extension is correctly tokenising the file and returning colours for tokens, but VS doesn't seem to be using those colours when rendering the text.

**Comment by samizzo** [comment]
In the meantime, why don't you give [HLSL Tools](https://github.com/tgjones/HlslTools) a try?

**Comment by Lex-DRL** [comment]
Oh, I just didn't know about that ext.
But it seems great. Thanks a lot!

**Comment by samizzo** [comment]
Yeah I think it's better than NShader actually. It has a lot of extra features like navigation and statement completion. I think the syntax colouring might not be quite as extensive as NShader but it works!

**Comment by blair0011** [comment]
I have a fix for this. It seems that the visxmanifest only allows MPF versions 12 & 13. I am on 16 (aka. Visual Studio 2017). If you open up the range to be [12.0,) and recompile everything works. I had to figure out this fix because HLSL Tools doesn't do glsl compute shaders well, and I really need something to work for those.

**Comment by samizzo** [comment]
Ahh thank you very much! I'll make this change and push out a new build.
